### ISSUE TITLE
[BUG] Plugin scripts are executed at random time intervals (every couple of minutes)

### ISSUE THREAD

**Issue by chrisidefix** [matryer/bitbar#263]
Plugins appear to be executed randomly every couple of minutes, rather than at their scheduled times. If I schedule a plugin to run every 15min, it will actually show an update in the menu bar every 15 min, but the actual script is executed randomly every couple of minutes in between. It appears the results of those "in between" script calls are discarded (at least I don't see any update in the MenuBar), but it's certainly a big problem for many application scenarios.

You can easily see how this can be a problem for triggering any action through these scheduled plugin scripts, which will likely decrease performance of your machine (or remote machine), if you only want to trigger a build every hour, but in fact start off one every 2-4 minutes (just as an example).

I only tested this with a Python plugin, but maybe someone can confirm that it in fact concerns all BitBar scripts? You can read about my specific issue here: https://github.com/matryer/bitbar-plugins/issues/296

Essentially, in this specific scenario (requesting weather data from forecast.io) it's an even bigger problem, since API calls (after 1000 calls) are not free. Scheduling these API calls through BitBar appears problematic if BitBar keeps calling the same script every 2-4 minutes, rather than every 15 minutes.

Of course I could add my own safety guards and check when the last API call really happened, but it seems that BitBar should be able to only call scripts at the specified intervals.

**Comment by chrisidefix** [comment]
I have tested this with simple bash scripts and could recreate the issue. It appears to occur after the computer has been asleep for a while. Here are some simple log entries:

bash script supposedly executed every 15 minutes:
```
Sun Mar  6 19:48:53 AEST 2016
Sun Mar  6 19:50:07 AEST 2016
Sun Mar  6 20:03:53 AEST 2016
Sun Mar  6 20:05:07 AEST 2016
Mon Mar  7 08:08:37 AEST 2016
Mon Mar  7 08:10:50 AEST 2016
Mon Mar  7 08:12:04 AEST 2016
```

**Comment by seripap** [comment]
+1 seeing discrepancies here as well @v1.6

**Comment by matryer** [comment]
Ok we'll look into this. Thanks for the reports guys.

**Comment by iosdeveloper** [comment]
This should be fixed as well by c352e7e and bfb47b3.
